### PR TITLE
poll program accounts

### PR DESCRIPTION
**Pull request by jhunsaker**
this is a partial change - the code to activate the polling is not included yet (but was tested locally)

this refactor should not change existing behavior, but allows for code deduplication

### CODE REVIEW

**Comment by jhunsaker** [comment]
additional follow up changes are to add filtering - this is grabbing all program accounts - including larger accounts that don't need to be grabbed as frequently

File: pc/rpc_client.hpp [hunk 1]
```diff
@@ -474,61 +470,64 @@ namespace pc
...
+      // parameters
+      void set_commitment( commitment );
+
+      account_subscribe();
+      void request( json_wtr& ) override;
+      void response( const jtree& ) override;
+      bool notify( const jtree& ) override;
+
+    private:
+      commitment  cmt_;
```
Comment (mass) [inline pc/rpc_client.hpp:493]: Should `cmt_` be moved into the base class seeing all the derived classes have it?

File: pc/rpc_client.hpp [hunk 2]
```diff
@@ -450,22 +452,16 @@ namespace pc
...
 
-      account_subscribe();
-      void request( json_wtr& ) override;
-      void response( const jtree& ) override;
-      bool notify( const jtree& ) override;
-
-    private:
-      pub_key    *acc_;
+    protected:
+      pub_key     acc_;
```
Comment (mass) [inline pc/rpc_client.hpp:456]: Why did this change from a `pub_key*` to a `pub_key`?

**Review by mass — APPROVED** [review]

File: pc/rpc_client.hpp (hunk 1 above)
Comment (jhunsaker) [reply pc/rpc_client.hpp:493]: yes, thanks, I noticed that but didn't want to repeat the testing for a small change - will include in the next round

File: pc/rpc_client.hpp (hunk 2 above)
Comment (jhunsaker) [reply pc/rpc_client.hpp:456]: because it's an output of the program calls whereas in the account_subscribe it's a parameter

**Pull request merged**